src/prefabs/monthPickerInput.tsx

```diff
@@ -0,0 +1,338 @@
+import * as React from 'react';
+import {
+  BeforeCreateArgs,
+  Icon,
+  prefab as makePrefab,
+  PrefabComponentOption,
+} from '@betty-blocks/component-sdk';
+import { DateTimePicker } from './structures/DateTimePicker';
+
+const beforeCreate = ({
+  close,
+  components: {
+    Content,
+    Field,
+    Footer,
+    Header,
+    FormField,
+    Toggle,
+    PropertySelector,
+    Label,
+    TextInput: Text,
+    CircleQuestion,
+    BBTooltip,
+  },
+  prefab: originalPrefab,
+  save,
+  helpers,
+}: BeforeCreateArgs) => {
+  const {
+    BettyPrefabs,
+    prepareInput,
+    useModelIdSelector,
+    useActionIdSelector,
+    usePrefabSelector,
+    usePropertyQuery,
+    setOption,
+    createUuid,
+    useModelQuery,
+    createBlacklist,
+    useModelRelationQuery,
+  } = helpers;
+
+  const [propertyPath, setProperty] = React.useState<any>('');
+  const [variableInput, setVariableInput] = React.useState(null);
+  const modelId = useModelIdSelector();
+  const actionId = useActionIdSelector();
+  const selectedPrefab = usePrefabSelector();
+  const [model, setModel] = React.useState<any>(null);
+  const [propertyBased, setPropertyBased] = React.useState(!!modelId);
+  const [prefabSaved, setPrefabSaved] = React.useState(false);
+
+  const [validationMessage, setValidationMessage] = React.useState('');
+
+  const modelRequest = useModelQuery({
+    variables: { id: modelId },
+    onCompleted: (result) => {
+      setModel(result.model);
+    },
+  });
+
+  const validate = () => {
+    if (modelRequest.loading) {
+      setValidationMessage(
+        'Model details are still loading, please try submitting again.',
+      );
+      return false;
+    }
+
+    return true;
+  };
+
+  let name: string | undefined;
+  let propertyKind;
+  let propertyModelId;
+  const componentId = createUuid();
+
+  function isProperty(path: string) {
+    return (
+      typeof path !== 'string' &&
+      typeof path === 'object' &&
+      !Array.isArray(path)
+    );
+  }
+
+  let propertyId: string;
+  if (isProperty(propertyPath)) {
```

Comment:
I would like to suggest to make this piece of code a function on its own.
```typescript
const getPropertyId = (propertyPath: <types>) => {
  if (isProperty(propertyPath)) {
    const { id } = propertyPath;
    return Array.isArray(id) ? id[id.length - 1] : id;
  }     
    return propertyPath;
}
```
And then change the `let` to `const`
...
```typescript
const propertyId = getPropertyId(propertyPath);
```

Reply:
This code is literally copied from `datePickerInput.tsx`. I agree with the suggestions and that the code quality can be improved, but unfortunately I don't have the time to refactor all the code that the product department originally wrote.